soc/posix/inf_clock/soc.h

```diff
@@ -43,7 +43,7 @@ void posix_soc_clean_up(void);
  * any Zephyr thread are running.
  */
 #define NATIVE_TASK(fn, level, prio)	\
-	static void (*_CONCAT(__native_task_, fn)) __used	\
+	static const void (*_CONCAT(__native_task_, fn)) __used	__noasan \
```

Comment:
could use an explanation of why 'const' is getting added here.

Reply:
Good point.  Will add a comment on the next respin.  The best reason is just hygine.  These are initialization records a-la SYS_INIT() (but for native_posix startup) and aren't ever mutated.

But the real reason is that marking them const is enough to get gcc to stop instrumenting them with padding.  Unfortunately clang wasn't so smart and needed that __noasan attribute.